actionview/lib/action_view/template/html.rb

```diff
@@ -26,9 +32,12 @@ def render(*args)
         to_str
       end
 
-      def formats
-        [@type.respond_to?(:ref) ? @type.ref : @type.to_s]
+      def format
+        @type
       end
+
+      def formats; Array(format); end
+      deprecate :formats
```

Comment:
Why should we do deprecation here in HTML and in Text? They’re both marked nodoc.

Reply:
tbh, I think we can remove them.  The problem is that these are *supposed* to quack the same as instances of `Template`.  I really don't think people are calling `formats` on anything but instances of `Template`, but I figured I'd add the deprecation warning since these are "duck typing buddies" with `Template`.

Actually I think these should probably share inheritance hierarchy with `Template`, but that's a later patch.

Reply:
(we were definitely calling `formats` on Text / HTML instances, and the deprecation warning helped me track / fix them)